rust/azure_iot_operations_connector/src/base_connector/managed_azure_device_registry.rs

```diff
@@ -795,132 +920,64 @@ impl AssetClient {
-                        // otherwise, only send an update if the dataset definition has changed
-                        if default_dataset_destination_updated
-                            || received_dataset_definition != dataset_definition
-                        {
-                            // we need to make sure we have the updated definition for comparing next time
-                            *dataset_definition = received_dataset_definition.clone();
-                            // send update to the dataset
-                            let _ = dataset_update_tx
-                                .send((
-                                    received_dataset_definition.clone(),
-                                    default_dataset_destinations.clone(),
-                                    self.release_dataset_notifications_tx.subscribe(),
-                                )).inspect_err(|tokio::sync::mpsc::error::SendError((e_dataset_definition, _,_))| {
-                                    // TODO: should this trigger the datasetClient create flow, or is this just indicative of an application bug?
-                                    log::warn!(
-                                        "Update received for dataset {} on asset {:?}, but DatasetClient has been dropped",
-                                        e_dataset_definition.name,
-                                        self.asset_ref
-                                    );
-                                });
+                    let connector_context_clone = self.connector_context.clone();
+                    let asset_ref_clone = self.asset_ref.clone();
+                    // Spawn a new task to prevent a possible cancellation and ensure the deleted
+                    // notification reaches the application.
+                    tokio::task::spawn(
+                        async move {
+                            Self::unobserve_asset(&connector_context_clone, &asset_ref_clone).await;
                         }
-                    }
-                    // it needs to be created
-                    else if let Err(e) = self.setup_new_dataset(
-                        received_dataset_definition.clone(),
-                        &default_dataset_destinations,
-                    ) {
-                        // If an error is returned, continue to process other datasets even if one isn't valid. Don't give this one to
-                        // the application since we can't forward data on it. If there's an update to the
-                        // definition, they'll get the create notification for it at that point if it's valid
-                        dataset_config_errors.push(e);
-                    }
+                    );
+                    // Asset update has been fully processed, mark as seen.
+                    self.asset_update_watcher_rx.mark_unchanged();
```

Comment:
no update would have been put into the channel in this flow - is this mark_unchanged() needed? Or is it just for cleanup safety so that we always continue to return deleted after this point? In that case, would it be better to close the watcher_tx channel?

Reply:
It is the first case, I can't close it without dropping the channel though